test/rect_test.py

```diff
@@ -2155,6 +2149,300 @@ def test_collidelistall_list_of_object_with_rect_property(self):
         ]
         self.assertFalse(r.collidelistall(f))
 
+    def test_collideobjects_call_variants(self):
+        # arrange
+        r = Rect(1, 1, 10, 10)
+        rects = [Rect(1, 2, 3, 4), Rect(10, 20, 30, 40)]
+        objects = [
+            self._ObjectWithMultipleRectAttribute(
+                Rect(1, 2, 3, 4), Rect(10, 20, 30, 40), Rect(100, 200, 300, 400)
+            ),
+            self._ObjectWithMultipleRectAttribute(
+                Rect(1, 2, 3, 4), Rect(10, 20, 30, 40), Rect(100, 200, 300, 400)
+            ),
+        ]
+
+        # act / verify
+        print("rects")
+        r.collideobjects(rects)
+        print("rects, key=None")
+        r.collideobjects(rects, key=None)
+        print("objects, key=lambda")
+        r.collideobjects(objects, key=lambda o: o.rect1)
+        print("assertRaises: objects")
+        self.assertRaises(TypeError, r.collideobjects, objects)
+
```

Comment:
There are some debug print statements left in the code here.

Reply:
removed, thanks for spotting